src/main/java/com/google/sps/servlets/FreeTimeServlet.java

```diff
@@ -0,0 +1,211 @@
+    // Initialize the freeTime utility. Keep track of the free time in the next 5 days, with
+    // work hours as defined between 10am and 6 pm. The rest of the time between 7 am and 11 pm
+    // should be considered personal time.
+    int personalBeginHour = 7;
+    int workBeginHour = 10;
+    int workEndHour = 18;
+    int personalEndHour = 23;
+    int numDays = 5;
+    FreeTimeUtility freeTimeUtility =
+        new FreeTimeUtility(
+            timeMin, personalBeginHour, workBeginHour, workEndHour, personalEndHour, numDays);
+    for (Event event : calendarEvents) {
+      DateTime start = event.getStart().getDateTime();
+      start = start == null ? event.getStart().getDate() : start;
+      DateTime end = event.getEnd().getDateTime();
+      end = end == null ? event.getEnd().getDate() : end;
+      Date eventStart = new Date(start.getValue());
+      Date eventEnd = new Date(end.getValue());
+      if (eventStart.before(timeMin)) {
+        eventStart = timeMin;
+      }
+      if (eventEnd.after(timeMax)) {
+        eventEnd = timeMax;
+      }
+      freeTimeUtility.addEvent(eventStart, eventEnd);
+    }
+
+    int wordCount = getWordCount(googleCredential);
+    int averageReadingSpeed = 50;
+    int minutesToRead = (int) Math.ceil((double) wordCount / averageReadingSpeed);
+    long timeNeeded = minutesToRead * TimeUnit.MINUTES.toMillis(1);
+    List<DateInterval> potentialTimes = getPotentialTimes(freeTimeUtility, timeNeeded);
+
+    PlanMailResponse planMailResponse =
+        new PlanMailResponse(wordCount, averageReadingSpeed, minutesToRead, potentialTimes);
+    // Convert event list to JSON and print to response
+    JsonUtility.sendJson(response, planMailResponse);
+  }
+
+  private List<DateInterval> getPotentialTimes(FreeTimeUtility freeTimeUtility, long timeNeeded) {
+    List<DateInterval> workFreeInterval = freeTimeUtility.getWorkFreeInterval();
+    List<DateInterval> potentialEventTimes = new ArrayList<>();
+    long remainingTime = timeNeeded;
+    for (DateInterval interval : workFreeInterval) {
+      Date potentialEnd = new Date(interval.getStart().getTime() + remainingTime);
+      if (interval.getEnd().before(potentialEnd)) {
+        remainingTime -= (interval.getEnd().getTime() - interval.getStart().getTime());
+        potentialEventTimes.add(interval);
+      } else {
+        potentialEventTimes.add(new DateInterval(interval.getStart(), potentialEnd));
+        break;
+      }
+    }
+    return potentialEventTimes;
+  }
+
+  private int getWordCount(Credential googleCredential) {
+    GmailClient gmailClient = gmailClientFactory.getGmailClient(googleCredential);
+    GmailClient.MessageFormat messageFormat = GmailClient.MessageFormat.FULL;
+    int numberDays = 7;
+    int wordCount = 0;
+    List<Message> unreadMessages = new ArrayList<>();
+    try {
+      unreadMessages = gmailClient.getUnreadEmailsFromNDays(messageFormat, numberDays);
+    } catch (IOException e) {
+      System.out.println(e);
```

Comment:
You probably want to wrap the exception in some custom type and re-throw it here. Otherwise the user will just think they have no unread e-mails to deal with.

Reply:
I used `Throwables` to re-throw the exceptions.